controllers/utils.go

```diff
@@ -470,3 +472,59 @@ func addRestoreLabelSelector(
 		)
 	}
 }
+
+// update BackupSchedule status and remove velero schedules
+// when the BackupSchedule is paused or in BackupCollision
+func updateBackupSchedulePhaseWhenPaused(
+	ctx context.Context,
+	c client.Client,
+	veleroScheduleList veleroapi.ScheduleList,
+	backupSchedule *v1beta1.BackupSchedule,
+	phase v1beta1.SchedulePhase,
+	msg string,
+) (ctrl.Result, error) {
+	scheduleLogger := log.FromContext(ctx)
+
+	if phase != v1beta1.SchedulePhasePaused && phase != v1beta1.SchedulePhaseBackupCollision {
+		// do not process any other type of schedule here
+		return ctrl.Result{}, nil
+	}
+
+	// delete schedules, so we don't generate new backups
+	for i := range veleroScheduleList.Items {
+		scheduleLogger.Info(
+			"Attempt to delete schedule " + veleroScheduleList.Items[i].Name,
+		)
+
+		if err := c.Delete(ctx, &veleroScheduleList.Items[i]); err == nil {
+			scheduleLogger.Info(
+				"Schedule deleted successfully " + veleroScheduleList.Items[i].Name,
+			)
+		} else {
+
+			// ignore not found errors
+			if !kerrors.IsNotFound(err) {
+				errMsg := "Failed to delete schedule " + veleroScheduleList.Items[i].Name
+				scheduleLogger.Error(err,
+					errMsg,
+				)
+				// update failed, retry
+				backupSchedule.Status.LastMessage = errMsg
+				c.Status().Update(ctx, backupSchedule)
+				return ctrl.Result{RequeueAfter: failureInterval}, err
+			}
+		}
+	}
+
+	//update status
+	backupSchedule.Status.Phase = phase
+	backupSchedule.Status.LastMessage = msg
+
+	backupSchedule.Status.VeleroScheduleCredentials = nil
+	backupSchedule.Status.VeleroScheduleManagedClusters = nil
+	backupSchedule.Status.VeleroScheduleResources = nil
+
+	c.Status().Update(ctx, backupSchedule)
```

Comment:
```suggestion
	err := c.Status().Update(ctx, backupSchedule)
	return ctrl.Result{}, err
```

Suggest we check for errors in case the status update fails, that way if we didn't manage to update the status to "paused", it'll retry and we'll get there.